test/lib/ufe/testAttribute.py

```diff
@@ -613,9 +644,139 @@ def testZAttrChangeRedoAfterPrimCreateRedo(self):
+        mayaUsd_createStageWithNewLayer.createStageWithNewLayer()
+        proxyShapes = cmds.ls(type="mayaUsdProxyShapeBase", long=True)
+        proxyShapePath = proxyShapes[0]
+        proxyShapeItem = ufe.Hierarchy.createItem(ufe.PathString.path(proxyShapePath))
+        proxyShapeContextOps = ufe.ContextOps.contextOps(proxyShapeItem)
+
+        stage = mayaUsd.lib.GetPrim(proxyShapePath).GetStage()
+
+        # create 3 sub-layers ( LayerA, LayerB, LayerC ) and set the edit target to LayerB.
+        rootLayer = stage.GetRootLayer()
+        subLayerC = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="SubLayerC")[0]
+        subLayerB = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="SubLayerB")[0]
+        subLayerA = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="SubLayerA")[0]
+
+        # check to see the sublayers added
+        addedLayers = [subLayerA, subLayerB, subLayerC]
+        self.assertEqual(rootLayer.subLayerPaths, addedLayers)
+
+        # set the edit target to LayerB
+        cmds.mayaUsdEditTarget(proxyShapePath, edit=True, editTarget=subLayerB)
+
+        # create a prim. This creates the primSpec in the SubLayerB.
+        proxyShapeContextOps.doOp(['Add New Prim', 'Sphere'])
+
+        spherePath = ufe.PathString.path('|stage1|stageShape1,/Sphere1')
+        sphereItem = ufe.Hierarchy.createItem(spherePath)
+        spherePrim = mayaUsd.ufe.ufePathToPrim(ufe.PathString.string(spherePath))
+
+        # create a transform3d
+        sphereT3d = ufe.Transform3d.transform3d(sphereItem)
+
+        # create a xformable
+        sphereXformable = UsdGeom.Xformable(spherePrim)
+
+        # writing to "transform op order" is expected to be allowed.
+        self.assertTrue(mayaUsdUfe.isAttributeEditAllowed(sphereXformable.GetXformOpOrderAttr()), True)
+
+        # do any transform editing.
+        sphereT3d = ufe.Transform3d.transform3d(sphereItem)
+        sphereT3d.scale(2.5, 2.5, 2.5)
+        sphereT3d.translate(4.0, 2.0, 3.0)
+
+        # check the "transform op order" stack.
+        self.assertEqual(sphereXformable.GetXformOpOrderAttr().Get(), Vt.TokenArray(('xformOp:translate', 'xformOp:scale')))
+
+        # check if translate attribute is editable
+        translateAttr = spherePrim.GetAttribute('xformOp:translate')
+        self.assertIsNotNone(translateAttr)
+
+        # authoring new transformation edit is expected to be allowed.
+        self.assertTrue(mayaUsdUfe.isAttributeEditAllowed(translateAttr), True)
+        sphereT3d.translate(5.0, 6.0, 7.0)
+        self.assertEqual(translateAttr.Get(), Gf.Vec3d(5.0, 6.0, 7.0))
+
+        # set the edit target to a weaker layer (LayerC)
+        cmds.mayaUsdEditTarget(proxyShapePath, edit=True, editTarget=subLayerC)
+
+        # authoring new transformation edit is not allowed.
+        self.assertFalse(mayaUsdUfe.isAttributeEditAllowed(translateAttr), False)
```

Comment:
Might want to write a value different from the existing one and confirm that the write did not happen.